### ISSUE TITLE
Setup CI for checking sanity of with_tests branch

### ISSUE THREAD

**Issue by martinsson** [martinsson/Refactoring-Kata-Lift-Pass-Pricing#21]
It'd be great to have a CI (travis, circleci, ..) to avoid systematically testing all PRs to the with_tests branch by hand.

Note: Possibly one could imagine to do the same for the master branch, but that'd require the tests to actually pass whereas now they fail in order for people to see where to start. Either way let's start with configuring a CI on the with_tests branch

**Comment by codecop** [comment]
Test branch could also always merge against master. there are conflicts we have to do something

**Comment by rolger** [comment]
Hi Johan

is this still an open issue? I'd like to implement a github-action for this issue. I'd start with Java

**Comment by codecop** [comment]
Yes please do. I can review your changes then

**Comment by rolger** [comment]
Alright mission completed :-) I need to prepare a PR. The changes are not just about adding the workflow action,  need to fix the mysql driver and the unit-test startup. 

You'll find the unit-test startup fix in this PR:
https://github.com/martinsson/Refactoring-Kata-Lift-Pass-Pricing/pull/47

I'd suggest to merge this PR first and then I'll create a new one based on this. Or do you have another suggestion?

**Comment by rolger** [comment]
By the way a csharp implementation of the workflow should be easy as soon as the Java actions are available. So I'd continue to implement the same for csharp.

**Comment by martinsson** [comment]
Great job. Indeed it's still an open issue

**Comment by rolger** [comment]
Just completed the csharp pipeline. So how should I split up the PRs? What do we want ot merge into the master branch?

**Comment by rolger** [comment]
> Just completed the csharp pipeline. So how should I split up the PRs? What do we want to merge into the master branch?

Why I'm asking? First, the changes can be merged into the master branch easily. The code will not be executed on master, because I added a filter. But the master and with_test branch are out of sync. So I believe that merging from master to with_test won't be easy at all. Second, I need two additional changes, which don't relate to the feature but must be fixed upfront:

1. The JUnit test should start the spark server only once
2. An upgrade of the mysql-connector-java version to 8.0.28 (see https://stackoverflow.com/questions/53131321/spring-boot-jdbc-javax-net-ssl-sslexception-closing-inbound-before-receiving-p/53136942#53136942)

**Comment by codecop** [comment]
Merge to master is good idea. It should be possible from with-tests. I recommend one large pr for with-tests branch. Merge is a separate task we can do afterwards. I believe there are also things missing on with-tests

**Comment by martinsson** [comment]
Hi how could I make this easier? By merging master into with_tests to make it up to date?

**Comment by codecop** [comment]
@martinsson Johan maybe first drop the PR's which want to bump versions - unless you want that. I usually deactivate Security scans in kata projects. Second please accept or reject other open pull requests.

**Comment by codecop** [comment]
@rolger could you also add actions for java-simple and csharp-simple when available. In fact everything should be tested, also Scala and PHP and such. I have many github integrations in https://github.com/codecop/Parrot-Refactoring-Kata/tree/github-build

**Comment by rolger** [comment]
@codecop  I'll have a look at them.

By the way CSharp needs an upgrade. The .Net version is not supported anymore.

**Comment by martinsson** [comment]
> @martinsson Johan maybe first drop the PR's which want to bump versions - unless you want that. I usually deactivate Security scans in kata projects. Second please accept or reject other open pull requests. I commented on PRs for that.
Good point

> Also there is a simplify C# branch, can it be brougt to master in separate folder like java-simple? That would be great.
I'll have a look at that

**Comment by martinsson** [comment]
Brilliant @rolger :)

**Comment by martinsson** [comment]
Done
